src/env-inl.h

```diff
@@ -325,6 +326,44 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline void Environment::print_sync_warn() const {
+  if (!warn_on_sync_)
+    return;
+
+  v8::HandleScope handle_scope(isolate_);
+  v8::Local<v8::StackTrace> stack =
+      v8::StackTrace::CurrentStackTrace(isolate_,
+                                        10,
+                                        v8::StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
+    v8::Local<v8::StackFrame> sf = stack->GetFrame(i);
+    node::Utf8Value fn_name_s(isolate_, sf->GetFunctionName());
+    node::Utf8Value sn_s(isolate_, sf->GetScriptName());
+    int ln = sf->GetLineNumber();
+    int cn = sf->GetColumn();
```

Comment:
Can you give the variables slightly more human-readable names?  The two ints can be const, by the look of it.